libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1186,6 +1246,8 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 		solAssert(!!valueComponentType, "");
 		if (!var.annotation().type)
 		{
+			autoTypeDeductionNeeded++;
```

Comment:
Does this need to be a counter (as opposed to just a boolean)?

Reply:
bool is totally sufficient; my initial thought required a counter - until I found a better implementation.